conduit/pipeline/pipeline.go

```diff
@@ -405,6 +431,16 @@ func (p *pipelineImpl) Stop() {
 	}
 }
 
+// WhyStopped returns nil if a context was never provided or the pipeline
+// is yet to have stopped. Otherwise, it returns the cause of the pipeline's
+// context cancellation.
+func (p *pipelineImpl) WhyStopped() error {
```

Comment:
Yes. I was motivated to elevate this capability in the API, but I'll remove it as it's not actually necessary once you know about cancellation causes.

Reply:
Upon further thought, I've changed my mind having remembered the original purpose. Consider the [conduit client](https://github.com/tzaffi/conduit/blob/7e88db3d2c9fea7afc600b0cee46f1f7c4e4fffd/pkg/cli/cli.go#L103). It may want to know why the pipeline stopped but it only has access to the `Pipeline interface`. In fact, instead of the current `defer pipeline.Stop()` call, we may want to have:

```go
def func() {
    pline.Stop()
    if !errors.Is(pline.WhyStopped(), pipeline.BecauseStopMethod) {
        logger.Error("unexpected conduit pipeline exit: %v", pline.WhyStopped())
    }
}()
```

Reply:
Upon further reflection, I'm liking a more streamlined interface as suggested in [this comment](https://github.com/algorand/conduit/pull/97/files#r1227096289).

So we get  rid of `Stop()`, but keep `WhyStopped()`, and also have the client be in charge of listening to interrupt signals and and issuing cancellations it nows about and can ignore using `WhyStopped()`.

Reply:
I didn't get ride of `Stop()`, but did get rid of `WhyStopped()`. I added more questions probing the possible resurrection of these to #100 

see: 1158f757ad8429bca46dc2ce96d7eee4fa77f639